src/cclib/parser/daltonparser.py

```diff
@@ -737,20 +754,36 @@ def extract(self, inputfile, line):
         # @    Final gradient norm:           0.000003746706
         # ...
         #
+        if "Final HF energy" in line and not (hasattr(self, "mpenergies") or hasattr(self, "ccenergies")):
+            self.metadata["theory"] = "HF"
```

Comment:
How about "methodology"? It's useful to separate the HF and post-HF methods, but I would do it on the writeCSX.